src/commands/update.js

```diff
@@ -0,0 +1,203 @@
+   * List the plugins that have updates.
+   *
+   * @param {Array<ToUpdatePlugin>} plugins the plugins to update
+   */
+  async __list (plugins) {
+    const columns = {
+      plugin: {
+        width: 10,
+        get: row => `${row.name}`
+      },
+      current: {
+        minWidth: 10,
+        get: row => `${row.currentVersion}`
+      },
+      latest: {
+        get: row => `${row.latestVersion}`
+      }
+    }
+
+    cli.table(plugins, columns)
+  }
+
+  /**
+   * Install the plugins that have updates.
+   *
+   * @private
+   * @param {Array<ToUpdatePlugin>} plugins the plugins to update
+   * @param {boolean} needsConfirm true to show confirmation prompt
+   */
+  async __install (plugins, needsConfirm) {
+    await this.__list(plugins)
+    let _doUpdate = true
+
+    this.log() // newline
+
+    if (needsConfirm) {
+      _doUpdate = await prompt(`Update ${plugins.length} plugin(s)?`)
+    }
+
+    if (_doUpdate) {
+      // install the plugins in sequence
+      for (const plugin of plugins) {
+        await this.config.runCommand('plugins:install', [plugin.name])
+      }
+    }
+  }
+
+  /**
+   * Install the plugins that have  updates, with an interactive install.
+   *
+   * @private
+   * @param {Array<ToUpdatePlugin>} plugins the plugins to update
+   */
+  async __interactiveInstall (plugins) {
+    const inqChoices = plugins
+      .map(plugin => { // map to expected inquirer format
+        return {
+          name: `${plugin.name}@${plugin.latestVersion}`,
+          value: plugin.name
+        }
+      })
+
+    const response = await inquirer.prompt([{
+      name: 'plugins',
+      message: 'Select plugins to update',
+      type: 'checkbox',
+      choices: inqChoices
+    }])
+
+    // install the plugins in sequence
+    for (const plugin of response.plugins) {
+      await this.config.runCommand('plugins:install', [plugin])
+    }
+  }
+
+  /**
+   * Process the plugins, determine if they need updates or warnings.
+   *
+   * @param {object} cliRoot the root folder of the cli
+   * @param {Array<string>} corePlugins package names of the core plugins
+   * @param {Array<InstalledPlugin>} installedPlugins
+   * @returns {Array<ToUpdatePlugin}
+   */
+  async __processPlugins () {
+    const corePlugins = this.config.pjson.oclif.plugins
+    const plugins = []
+
+    // Filter installed plugins:
+    // - remove any plugin that is in core, that is not from the @adobe namespace
+    // These will not be updateable for compatibility reasons
+    const installedPlugins = this.config.plugins.filter(plugin =>
+      !(
+        corePlugins.includes(plugin.name) &&
+        !(plugin.name.startsWith('@adobe/'))
+      )
+    )
+
+    for (const plugin of installedPlugins) {
+      const { type, name, version: currentVersion } = plugin
+      const latestVersion = await getNpmLatestVersion(name)
+      let coreVersion = (type === 'core') ? currentVersion : null
+      const needsUpdate = (latestVersion > currentVersion)
```

Comment:
I think we probably need to go full semver here and not just text compare, there are numerous edge cases which I don't see covered by tests. ie. `v1.0.0` or `2.8.3-lts`
https://www.npmjs.com/package/semver

Reply:
good catch, i'll fix

Reply:
@purplecabbage PR updated